### PR TITLE
Jirka kbc 1513 create typed table doc

### PR DESCRIPTION
**Pull request by fejjo**
Jira: KBC-1513

Before asking for review make sure that:

- [x] New client method(s) has tests
- [x] Apiary file is updated
- [x] You declared if there is a BC break or not (will affect next release of a client)

---

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1545,6 +1545,106 @@ Lists tables in a given bucket.
                 }
             ]
 
+## Create table definition [/v2/storage/buckets/{bucket_id}/tables-definition]
+
+### Create new table definition [POST]
+
```
Comment (fejjo) [inline apiary.apib:1551]: Creates a new table in a bucket asynchronously by definition of table.

File: apiary.apib [hunk 2]
```diff
@@ -1545,6 +1545,106 @@ Lists tables in a given bucket.
...
+Creates a table with typed columns. *Synapse only.*
++ Parameters
+    + bucket_id (required) - The bucket Id
+
++ Attributes
+    + name (required) - New table name
+    + primaryKeysNames[] (string) - has to be subset of column names
+    + columns[] (required)
+       + name
```
Comment (fejjo) [inline apiary.apib:1562]: name ma byt required a string

File: apiary.apib [hunk 3]
```diff
@@ -1545,6 +1545,106 @@ Lists tables in a given bucket.
...
+       + definition (object)
+            + type (required) - datatype of this column, choose one from https://bit.ly/3uZjN60
```
Comment (fejjo) [inline apiary.apib:1564]: string

File: apiary.apib [hunk 4]
```diff
@@ -1545,6 +1545,106 @@ Lists tables in a given bucket.
...
+
+**EXPERIMENTAL**
+
+Creates a table with typed columns. *Synapse only.*
++ Parameters
+    + bucket_id (required) - The bucket Id
+
++ Attributes
+    + name (required) - New table name
+    + primaryKeysNames[] (string) - has to be subset of column names
```
Comment (fejjo) [inline apiary.apib:1560]: Toto moze byt optional?

File: apiary.apib (hunk 1 above)
Comment (fejjo) [reply apiary.apib:1551]: PRidal by som nejaku veticku co ten endpoint robi

File: apiary.apib [hunk 5]
```diff
@@ -1545,6 +1545,106 @@ Lists tables in a given bucket.
...
+
+            {
+                "name": "table_name",
+                "primaryKeysNames": [],
+                "columns": [
+                    {
+                        "name": "id",
+                        "definition": {
+                            "type": "INT",
+                            "length": "10",
```
Comment (fejjo) [inline apiary.apib:1591]:
```suggestion
                            "length": null,
```
pre int to nieje validna hodnota
Screenshot 2021-05-17 at 16 16 27

File: apiary.apib (hunk 4 above)
Comment (jirkasemmler) [reply apiary.apib:1560]: optional ne, ale prazdne ano, doplnim

File: apiary.apib [hunk 6]
```diff
@@ -1545,6 +1545,109 @@ Lists tables in a given bucket.
...
+       + name (required, string) - column name
+       + definition (object) - column settings
+            + type (required, string) - datatype of this column, choose one from https://bit.ly/3uZjN60
+            + length (string) - optional, default: depends on type
+            + nullable (boolean) - optional, default: true
+            + default (string) - optional, default value for column
+    + distribution
+        + type (required, enum[string])
+            + Members
+                + HASH - requires just on column in `distributionColumnsNames`
```
Comment (zajca) [inline apiary.apib:1574]:
```suggestion
                + HASH - requires exactly ono column in `distributionColumnsNames`
```

File: apiary.apib [hunk 7]
```diff
@@ -1545,6 +1545,109 @@ Lists tables in a given bucket.
...
+            },
+            "metrics": {
+                "inCompressed": false,
+                "inBytes": 0,
+                "inBytesUncompressed": 0,
+                "outCompressed": false,
+                "outBytes": 0,
+                "outBytesUncompressed": 0
+            }
+        }
```
Comment (zajca) [inline apiary.apib:1649]: Ta json response neodpovídá tomu co reálně ten endpoint vrací.
```suggestion
        {
   "id":255,
   "status":"waiting",
   "url":"https:\/\/localhost:8700\/v2\/storage\/jobs\/255",
   "tableId":null,
   "operationName":"tableDefinitionCreate",
   "operationParams":{
      "name":"test",
      "columns":[
         {
            "name":"test",
            "definition":{
               "type":"INT",
               "nullable":true
            }
         }
      ],
      "distribution":{
         "type":"ROUND_ROBIN",
         "distributionColumnsNames":[
            
         ]
      },
      "queue":"main"
   },
   "createdTime":"2021-05-24T08:40:47+0200",
   "startTime":null,
   "endTime":null,
   "runId":"1480",
   "results":null,
   "creatorToken":{
      "id":"2260",
      "description":"devel+test-common_admin@keboola.com"
   },
   "metrics":{
      "inCompressed":false,
      "inBytes":0,
      "inBytesUncompressed":0,
      "outCompressed":false,
      "outBytes":0,
      "outBytesUncompressed":0
   }
}
```

**Review by zajca — CHANGES_REQUESTED** [review]
Je to cajk, ale response nesedí, nesedí params ani název operace.

**Comment by jirkasemmler** [comment]
@zajca opraveno, jeste jsem to potunil aby to opravdu odpovidalo

**Review by zajca — APPROVED** [review]
Cajk :heavy_check_mark:

**Pull request merged**